src/UrlMatcher.php

```diff
@@ -94,19 +96,21 @@ public function match(ServerRequestInterface $request): MatchingResult
         }
 
         $dispatchData = $this->getDispatchData();
-        $path = urldecode($request
-            ->getUri()
-            ->getPath());
+        $path = urldecode($request->getUri()->getPath());
+        $uriPrefix = $this->urlGenerator->getUriPrefix();
```

Comment:
Can we pass prefix and not the whole URL generator?

Reply:
Without dependency? Nope.